lib/ansible/modules/storage/purestorage/purefa_volume.py

```diff
@@ -125,6 +154,24 @@ def create_volume(module, array):
     module.exit_json(changed=True)
 
 
+def copy_from_volume(module, array):
+    """Create Volume Clone"""
+    if not module.check_mode:
```

Comment:
IMHO this condition is on the wrong place, because the in check mode it seems not to return `changed=True` handled later in the code. As a result, a user may get a changed=False in check mode but a changed=True in live mode.

This condition should be right before the actual action(s) happens.